src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/errors/get_columns.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { EuiBasicTableColumn } from '@elastic/eui';
+import { EuiText, EuiTextTruncate, EuiLink } from '@elastic/eui';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import type { ErrorData, ErrorsByTraceId } from '@kbn/apm-types';
+import { TRACE_ID, SPAN_ID, PROCESSOR_EVENT, EVENT_NAME, ERROR_ID } from '@kbn/apm-types';
+import type { GenerateDiscoverLink } from '../../hooks/use_get_generate_discover_link';
+import { NOT_AVAILABLE_LABEL } from '../../common/constants';
+
+const getErrorMessage = (error: ErrorData) => {
+  if (error?.exception?.message) {
+    return error.exception.message;
+  }
+
+  if (error?.log?.message) {
+    return error.log.message;
+  }
+
+  return NOT_AVAILABLE_LABEL;
+};
+
+export const getColumns = ({
+  generateDiscoverLink,
+  traceId,
+  docId,
+  source,
+}: {
+  generateDiscoverLink: GenerateDiscoverLink;
+  traceId: string;
+  docId?: string;
+  source: string;
+}): Array<EuiBasicTableColumn<ErrorsByTraceId['traceErrors'][0]>> => [
+  {
+    field: 'name',
+    name: i18n.translate(
+      'unifiedDocViewer.observability.traces.docViewerSpanOverview.errors.table.error',
+      { defaultMessage: 'Error message and culprit' }
+    ),
+    sortable: (item) => item.error?.exception?.message || '',
+    render: (_, item) => {
+      const href = generateDiscoverLink({
+        [TRACE_ID]: traceId,
+        ...(docId && { [SPAN_ID]: docId }),
```

Comment:
`docId` could be either a **span.id** or a **transaction.id**, right?

Reply:
ah that's right. I have to update all the queries or simply pass the `spanId` instead or docId as I haven’t found any error document that contains a transaction.id but not a span.id. 

```
processor.event: "error" and NOT span.id: * and trace.id: * and transaction.id: *
```
and the unprocessed otel data will not have  `transaction.id` 

Am I missing something else here?

Reply:
ah that's looks good then. Makes sense.